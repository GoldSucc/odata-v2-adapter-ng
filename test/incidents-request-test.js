"use strict";

const supertest = require("supertest");

const env = require("./_env");
const util = require("./_env/util");

let request;

describe("incidents-request", () => {
  beforeAll(async () => {
    const context = await env("incidentsmodel", 0);
    request = supertest(context.app);
  });

  afterAll(async () => {
    await env.end();
  });

  it("GET incidents service", async () => {
    const response = await util.callRead(request, "/v2/incidents", {
      accept: "application/json",
    });
    expect(response.body).toBeDefined();
    expect(response.body.d.EntitySets.sort()).toMatchSnapshot();
  });

  it("GET incidents service data", async () => {
    const response = await util.callRead(request, "/v2/incidents/Incidents", {
      accept: "application/json",
    });
    expect(response.body).toBeDefined();
    expect(response.body.d.results.length > 0).toEqual(true);
  });
});
