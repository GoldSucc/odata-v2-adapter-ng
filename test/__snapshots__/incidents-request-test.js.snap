// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incidents-request GET incidents service 1`] = `
Array [
  "Category",
  "Category_texts",
  "Country",
  "Country_texts",
  "Currency",
  "Currency_texts",
  "Incidents",
  "Priority",
  "Priority_texts",
]
`;
